### PR TITLE
87085: Add logic for JSON <-> PDF validation

### PR DESCRIPTION
**Pull request by balexandr**
As a CHAMPVA applicant, I need my form data to populate correctly in the PDFs sent to the back office for processing.

## Summary

- Port over `pdf_matcher` from `ClaimsAPI` and update to remove "Signed electronically..." as the date will always be when the test is run so matching would always fail.
- Add PDFs to spec to use as the fully filled-out PDFs
- Update pdf_filler Rspec to use the new pdf_matcher
- Updated fixture JSON with missing values
- Updated form_mappings to fix mistake

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/87085

## Testing done

- [x] *Rspec*
- [x] *Manual viewing of PDFs before saving them as the ones to check against

### CODE REVIEW

File: modules/ivc_champva/app/form_mappings/vha_10_7959c.json.erb [hunk 1]
```diff
@@ -9,9 +9,9 @@
   "form1[0].#subform[0].applicantState2[0]":                                  "<%= form.data.dig('applicant_address', 'state') %>",
   "form1[0].#subform[0].applicantZipCode2[0]":                                "<%= form.data.dig('applicant_address', 'postal_code') %>",
   "form1[0].#subform[0].PhoneNumb-1[0]":                                      "<%= form.data.dig('applicant_phone') %>",
-  "form1[0].#subform[0].#area[4].applicantSexRadioButtons2[0]":               "<%= form.data.dig('applicant') == 'M' ? 0 : 1 %>", 
+  "form1[0].#subform[0].#area[4].applicantSexRadioButtons2[0]":               "<%= form.data.dig('applicant') == 'M' ? 0 : 1 %>",
   "form1[0].#subform[0].#area[0]":                                            "<%= form.data.dig('') %>",
-  "form1[0].#subform[0].NewAddressBox-1[0]":                                  "<%= form.data.dig('applicant_new_address') != 'no' ? 1 : 0 %>", 
+  "form1[0].#subform[0].NewAddressBox-1[0]":                                  "<%= form.data.dig('applicant_new_address') != 'no' ? 1 : 0 %>",
   "form1[0].#subform[0].PartABDBkgrnd[0].PartARadioButtonList[0]":            "<%= form.data.dig('applicant_medicare_status') ? 0 : 1 %>",
```
Comment (balexandr) [inline modules/ivc_champva/app/form_mappings/vha_10_7959c.json.erb:15]: Trailing spaces, VSCode automagically removed them

File: modules/ivc_champva/app/form_mappings/vha_10_7959c.json.erb [hunk 2]
```diff
@@ -43,16 +43,16 @@
...
-  "form1[0].#subform[0].Date-TermnNameInsurance-2[0]":                        "<%= form.data.dig('applicant_secondary_expiration_date') %>",  
+  "form1[0].#subform[0].Date-TermnNameInsurance-2[0]":                        "<%= form.data.dig('applicant_secondary_expiration_date') %>",
   "form1[0].#subform[0].RadioButtonList[3]":                                  "<%= form.data.dig('applicant_secondary_through_employer') ? 0 : 1 %>",
   "form1[0].#subform[0].RadioButtonList[4]":                                  "<%= form.data.dig('applicant_secondary_eob') == 'hasEob' ? 0 : 1 %>",
   "form1[0].#subform[0].RadioButtonList[5]":                                  "<%= form.data.dig('applicant_secondary_has_prescription') ? 0 : 1 %>",
-  "form1[0].#subform[0].Signature[0]":                                        "<%= form.data['signature'] %>",
+  "form1[0].#subform[0].Signature[0]":                                        "<%= form.data['statement_of_truth_signature'] %>",
   "form1[0].#subform[0].DateTimeField1[0]":                                   "<%= form.data.dig('certification_date') %>",
```
Comment (balexandr) [inline modules/ivc_champva/app/form_mappings/vha_10_7959c.json.erb:56]: @michaelclement is going to normalize `statement_of_truth_signature` in vets-website

File: modules/ivc_champva/spec/support/pdf_matcher.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'rspec/expectations'
+
+RSpec::Matchers.define :match_pdf_content_of do |expected|
+  match do |actual|
+    actual_reader = PDF::Reader.new(actual)
+    actual_pages = actual_reader.pages.size
+    actual_text = actual_reader.pages.map(&:text)
+    actual_text = actual_text.map do |line|
```
[the remark this answers is not in the document]
Comment (cloudmagic80) [reply modules/ivc_champva/spec/support/pdf_matcher.rb:10]: Is there a screenshot of the pdf matcher result?

[the remark this answers is not in the document]
Comment (balexandr) [reply modules/ivc_champva/spec/support/pdf_matcher.rb:10]: @cloudmagic80 No it just turns the PDFs into a single slot Array and then compares the two Arrays and returns T or F.

**Review by cloudmagic80 — APPROVED** [review]
Looks good

**Pull request merged**